src/views/LimitOrders/index.tsx

```diff
@@ -0,0 +1,428 @@
+        throw new Error('No account')
+      }
+      const inputToken = currencies.input instanceof Token ? wrappedCurrencies.input?.address : GELATO_NATIVE
+      const outputToken = currencies.output instanceof Token ? wrappedCurrencies.output?.address : GELATO_NATIVE
+
+      const orderToSubmit = {
+        inputToken,
+        outputToken,
+        inputAmount: rawAmounts.input,
+        outputAmount: rawAmounts.output,
+        owner: account,
+      }
+      console.log(orderToSubmit)
+      handleLimitOrderSubmission(orderToSubmit)
+        .then(({ hash }) => {
+          console.log('Order submitted!')
+          setSwapState((prev) => ({
+            attemptingTxn: false,
+            tradeToConfirm: prev.tradeToConfirm,
+            swapErrorMessage: undefined,
+            txHash: hash,
+          }))
+        })
+        .catch((error) => {
+          console.log('Submission failed!', error)
+          setSwapState((prev) => ({
+            attemptingTxn: false,
+            tradeToConfirm: prev.tradeToConfirm,
+            swapErrorMessage: error.message,
+            txHash: undefined,
+          }))
+        })
+    } catch (error) {
+      console.error(error)
+    }
+  }, [
+    handleLimitOrderSubmission,
+    account,
+    rawAmounts.input,
+    rawAmounts.output,
+    currencies.input,
+    currencies.output,
+    wrappedCurrencies.input?.address,
+    wrappedCurrencies.output?.address,
+  ])
+
+  const [showConfirmModal] = useModal(
+    <ConfirmLimitOrderModal
+      currencies={currencies}
+      formattedAmounts={formattedAmounts}
+      currentMarketRate={currentMarketRate?.toSignificant(4)}
+      currentMarketRateInverted={currentMarketRate?.invert().toSignificant(4)}
+      limitPrice={formattedAmounts.price}
+      limitPriceInverted={price?.invert().toSignificant(6)}
+      percentageRateDifference={parseFloat(percentageRateDifference?.toSignificant(3)).toLocaleString(undefined, {
+        minimumFractionDigits: 0,
+        maximumFractionDigits: 3,
+      })}
+      onConfirm={handlePlaceOrder}
+      attemptingTxn={attemptingTxn}
+      txHash={txHash}
+      customOnDismiss={handleConfirmDismiss}
+      swapErrorMessage={swapErrorMessage}
+    />,
+    true,
+    true,
+    'confirmLimitOrderModal',
+  )
+
+  // mark when a user has submitted an approval, reset onTokenSelection for input field
+  useEffect(() => {
+    if (approvalState === ApprovalState.PENDING) {
+      setApprovalSubmitted(true)
+    }
+  }, [approvalState, approvalSubmitted])
+
+  const showApproveFlow =
+    !inputError && (approvalState === ApprovalState.NOT_APPROVED || approvalState === ApprovalState.PENDING)
+
+  const inputCurrencyId =
+    currencies.input instanceof Token ? currencies.input.address : currencies.input === ETHER ? 'BNB' : ''
+  const outputCurrencyId =
+    currencies.output instanceof Token ? currencies.output.address : currencies.output === ETHER ? 'BNB' : ''
+
+  const userHasOrders = useMemo(
```

Comment:
if we keep this logic, the performance issue isn't easy to solve. The LimitOrder will keep re-rendering because of the interval. 

I would suggest we show the open table even when user has no order

Reply:
Thanks for the fix @chefilip 🙌